src/crimson/osd/replicated_backend.cc

```diff
@@ -38,31 +40,337 @@ ReplicatedBackend::_read(const hobject_t& hoid,
   return store->read(coll, ghobject_t{hoid}, off, len, flags);
 }
 
+
+/*
+ * High level transaction.
+ *
+ * Due to how clones, renames (and EC) are handled in RADOS, crimson
+ * needs something similar to PGTransaction. Unlike it, HLTransaction
+ * is just an extension (also from the memory arrangement's POV) to
+ * the low-level os::Transaction. These jettisonable memory areas
+ * helps to reorder operations.
+ */
+namespace crimson::osd {
+class HLTransaction {
```

Comment:
The osd_op from the client is higher level still.  How about PGBackendTransaction (or PGBackend::Transaction) -- that way readers will quickly associate this structure with the interface that consumes it.